# Reddit ADHD Women Message Board Comments: EDA

In [4]:
# ADHD Reddit Women
import pandas as pd
adhdw = pd.read_csv(r'C:\Career\LighthouseLabs\Final Project\Potential Datasets\ADHD Reddit\adhdwomen-comment.csv')

In [2]:
adhdw.shape

(202658, 5)

In [3]:
high_scoresw_50 = adhdw.loc[adhdw['score'] > 50, 'body']
print(high_scoresw_50)

2480      Like, outside of context you may be right. But...
20617     Actually lots of folks failed at college becau...
20720     I did full psycho educational testing as part ...
52103     There are people way more fucked up in the hea...
52129     That psychiatrist is full of shit. I'm so sorr...
                                ...                        
202236    I have a very sensitive gag reflex, so my spou...
202255    I bought a magnetic whiteboard for the front o...
202265    I read somewhere that if plastic containers ha...
202632    The book is called\n\n\n You Mean I'm Not Lazy...
202646    Ugh, I have the other part - where the 'gifted...
Name: body, Length: 338, dtype: object


In [4]:
high_scoresw_50.shape

(338,)

In [5]:
high_scoresw_100 = adhdw.loc[adhdw['score'] > 100, 'body']
print(high_scoresw_100)

54025     Every time I read something like this I think ...
54497     I got that from a guy who goes to Burning Man ...
55239     I used to read fiction books in school-I would...
55309     &gt;"because you were able to complete a Maste...
56445     honestly, can text messages get a 'remind me' ...
                                ...                        
169448    This was a MAJOR realization of mine recently ...
200892    Sounds like it could be sensory overload? Ther...
200939    Oooh I find the opposite! If I can feel my und...
201007    The biggest lie we tell ourselves is, "I will ...
202646    Ugh, I have the other part - where the 'gifted...
Name: body, Length: 101, dtype: object


In [6]:
high_scoresw_100.shape

(101,)

In [9]:
high_scoresw_150 = adhdw.loc[adhdw['score'] > 150, 'body']
print(high_scoresw_150)

56445     honestly, can text messages get a 'remind me' ...
76082     Without electronic stimulation or diversion I ...
77363     *avoid entire rooms in my house for days at a ...
77715     Yup. I work as a cashier. Imagine the pain of ...
79163     Woah. I didn’t realize that’s why I did that 🤯...
108255    every single day. i think it is because i’m ov...
109156    This is a very real thing! ADHD coaches recomm...
109756    I would heavily follow the "treat others the w...
110944    Umm...How did I never know I needed this until...
111579    This is why I hate all these scenarios where I...
128129    So my psychologist explained that some people ...
128598    Binge eating as a means of dopamine seeking is...
140715    Are you bloody joking me? Normal people can re...
140727    My neurotypical partner says no, so I’m really...
142189    Thank you. This was almost too real to read, b...
142208    I hear about too many highly intelligent peopl...
143341    Also we have some kind of memo

In [10]:
high_scoresw_150.shape

(43,)

In [11]:
high_scoresw_200 = adhdw.loc[adhdw['score'] > 200, 'body']
print(high_scoresw_200)

76082     Without electronic stimulation or diversion I ...
77363     *avoid entire rooms in my house for days at a ...
79163     Woah. I didn’t realize that’s why I did that 🤯...
128598    Binge eating as a means of dopamine seeking is...
140715    Are you bloody joking me? Normal people can re...
140727    My neurotypical partner says no, so I’m really...
143341    Also we have some kind of memory glitch where ...
143775    They forgot regret. Outrage when you find out ...
146611    Lol I didn't know other people did this. I pro...
148091    The self awareness is probably the hardest sym...
153195    Good job! You are only blessing yourself each ...
154488    They're using it to work 12 hours straight, me...
154532    This so so so much. \n\nAdderall kicks in, nap...
155029    I think it's called sleep revenge. You spend a...
167307    Forced prioritizing! I also thrive best when m...
168300    😭\n\nI don't know who you are but you've descr...
168304    Ok but seriously, that asshole

In [12]:
high_scoresw_200.shape

(20,)

In [25]:
# Low Scores (downvoted) Women
low_scoresw_neg8 = adhdw.loc[adhdw['score'] < -8, 'body']
print(low_scoresw_neg8)

54974                                             [removed]
56620     can you send some pics? Nude? Just fpr researc...
62255                                        #Don't take it
74504     As someone who is diagnosed with "severe adult...
84285     So am I going to get banned from this sub too ...
84308     I’m comfortable voting as I plan to. Just wond...
84401     If you don't want kids then don't have unprote...
109733                                            [removed]
111657    Not a woman and don't have ADHD (front page al...
144453    This is r/adhdwomen so you can probably just u...
144462    NBs aren’t women and most women just go by she...
144479    I may not be NB, but I am a privileged white g...
149681    You can't change a sleep pattern by going to b...
150057    Well, I don't know that you can write a post t...
152099                                                   No
155458    Thank you, but for the questionably homophobic...
198857    Less pretending, and more work

In [26]:
low_scoresw_neg8.shape

(17,)

In [29]:
print(adhdw['body'][152099])
print(adhdw['score'][152099])

No
-19


VADER Sentiment Analysis with Sentiment Intensity Analyzer

In [1]:
import nltk
from nltk.corpus import stopwords
from nltk.tokenize import word_tokenize
from gensim.models import Word2Vec
from nltk.sentiment.vader import SentimentIntensityAnalyzer
import pandas as pd

In [ ]:
# Already Done or Unneeded Steps from Google Colab analysis
# # Import data before pre-processing
# dfw = pd.read_csv('adhdwomen-comment.csv')
# dfw = pd.read_csv('adhdwomen-comment.csv', encoding='utf-8')
# # Google Colab necessity
# !pip install nltk  # Install nltk if not already installed
# import nltk
# nltk.download('stopwords')

In [2]:
# Step 1: Prepare your text data
stop_words = set(stopwords.words('english'))

In [ ]:
# More Google Colab necessity
# !pip install nltk  # Install nltk if not already installed
# import nltk
# nltk.download('punkt')

In [5]:
# Step 2: Tokenize your text
def tokenize_text(text):
    tokens = word_tokenize(text)
    tokens = [token.lower() for token in tokens if token.lower() not in stop_words and token.isalpha()]
    return tokens

adhdw['tokens'] = adhdw['body'].apply(tokenize_text)

In [6]:
# Step 3: Train a Word2Vec model
word2vec_model = Word2Vec(sentences=adhdw['tokens'], vector_size=100, window=5, min_count=1, workers=4)

In [7]:
# Step 4: Generate word embeddings
def generate_word_embeddings(tokens):
    embeddings = []
    for token in tokens:
        if token in word2vec_model.wv.key_to_index:
            embeddings.append(word2vec_model.wv[word2vec_model.wv.key_to_index[token]])
    return embeddings

adhdw['embeddings'] = adhdw['tokens'].apply(generate_word_embeddings)

In [8]:
# Step 5: Aggregate the word embeddings
def aggregate_embeddings(embeddings):
    if len(embeddings) > 0:
        aggregated_embedding = sum(embeddings) / len(embeddings)
    else:
        aggregated_embedding = []
    return aggregated_embedding

adhdw['aggregated_embedding'] = adhdw['embeddings'].apply(aggregate_embeddings)

In [9]:
# Time for VADER
import nltk
nltk.download('vader_lexicon')

[nltk_data] Downloading package vader_lexicon to
[nltk_data]     C:\Users\User\AppData\Roaming\nltk_data...
[nltk_data]   Package vader_lexicon is already up-to-date!


True

In [11]:
# Step 6: Apply sentiment analysis using VADER Intensity Analyzer
analyzer = SentimentIntensityAnalyzer()

def analyze_sentiment(embedding):
    embedding_str = [str(item) for item in embedding]
    sentiment_scores = analyzer.polarity_scores(" ".join(embedding_str))
    return sentiment_scores

adhdw['sentiment_scores'] = adhdw['tokens'].apply(analyze_sentiment)

In [12]:
print(adhdw['sentiment_scores'])

0         {'neg': 0.132, 'neu': 0.575, 'pos': 0.293, 'co...
1         {'neg': 0.118, 'neu': 0.743, 'pos': 0.139, 'co...
2         {'neg': 0.169, 'neu': 0.637, 'pos': 0.195, 'co...
3         {'neg': 0.0, 'neu': 1.0, 'pos': 0.0, 'compound...
4         {'neg': 0.023, 'neu': 0.788, 'pos': 0.188, 'co...
                                ...                        
202653    {'neg': 0.0, 'neu': 0.39, 'pos': 0.61, 'compou...
202654    {'neg': 0.0, 'neu': 0.648, 'pos': 0.352, 'comp...
202655    {'neg': 0.264, 'neu': 0.536, 'pos': 0.2, 'comp...
202656    {'neg': 0.185, 'neu': 0.645, 'pos': 0.169, 'co...
202657    {'neg': 0.0, 'neu': 0.86, 'pos': 0.14, 'compou...
Name: sentiment_scores, Length: 202658, dtype: object


In [15]:
# Accessing sentiment scores for 167307 document, a Top 20 highest score
scores = adhdw['sentiment_scores'][167307]
negative_score = scores['neg']
neutral_score = scores['neu']
positive_score = scores['pos']
compound_score = scores['compound']

# Printing the scores
print("Negative score:", negative_score)
print("Neutral score:", neutral_score)
print("Positive score:", positive_score)
print("Compound score:", compound_score)
print(adhdw['body'][167307])


Negative score: 0.169
Neutral score: 0.443
Positive score: 0.388
Compound score: 0.8126
Forced prioritizing! I also thrive best when my choices are made for me. Not trying to be smart, that's literally how it is. Learning to appropriate prioritize stuff is the one skill I feel like I could probably improve rather than accept I'm just bad at.


In [16]:
# Accessing sentiment scores for 153195 document, a Top 20 highest score
scores = adhdw['sentiment_scores'][153195]
negative_score = scores['neg']
neutral_score = scores['neu']
positive_score = scores['pos']
compound_score = scores['compound']

# Printing the scores
print("Negative score:", negative_score)
print("Neutral score:", neutral_score)
print("Positive score:", positive_score)
print("Compound score:", compound_score)
print(adhdw['body'][153195])

Negative score: 0.059
Neutral score: 0.523
Positive score: 0.418
Compound score: 0.8836
Good job! You are only blessing yourself each time you brush. I am so happy you are making this a priority. Most ADHD medications are really hard on the mouth, so you are even helping yourself out even more so. 

Signed,
Dental Professional With ADHD


In [30]:
# Accessing sentiment scores for 152099 document, a Top 20 downvotes (-19) score
scores = adhdw['sentiment_scores'][152099]
negative_score = scores['neg']
neutral_score = scores['neu']
positive_score = scores['pos']
compound_score = scores['compound']

# Printing the scores
print("Negative score:", negative_score)
print("Neutral score:", neutral_score)
print("Positive score:", positive_score)
print("Compound score:", compound_score)
print(adhdw['body'][152099])

Negative score: 0.0
Neutral score: 0.0
Positive score: 0.0
Compound score: 0.0
No


In [31]:
# Accessing sentiment scores for 56620 document, a Top 20 downvotes (-18) score
scores = adhdw['sentiment_scores'][56620]
negative_score = scores['neg']
neutral_score = scores['neu']
positive_score = scores['pos']
compound_score = scores['compound']

# Printing the scores
print("Negative score:", negative_score)
print("Neutral score:", neutral_score)
print("Positive score:", positive_score)
print("Compound score:", compound_score)
print(adhdw['body'][56620])
print("Downvotes:", adhdw['score'][56620])

Negative score: 0.0
Neutral score: 1.0
Positive score: 0.0
Compound score: 0.0
can you send some pics? Nude? Just fpr research purposses
Downvotes: -18


In [32]:
# Accessing sentiment scores for 198857 document, a Top 20 downvotes (-13) score
scores = adhdw['sentiment_scores'][198857]
negative_score = scores['neg']
neutral_score = scores['neu']
positive_score = scores['pos']
compound_score = scores['compound']

# Printing the scores
print("Negative score:", negative_score)
print("Neutral score:", neutral_score)
print("Positive score:", positive_score)
print("Compound score:", compound_score)
print(adhdw['body'][198857])
print("Downvotes:", adhdw['score'][198857])

Negative score: 0.0
Neutral score: 0.568
Positive score: 0.432
Compound score: 0.4228
Less pretending, and more work, might actually help your life.
Downvotes: -13


In [33]:
# Accessing sentiment scores for 84285 document, a Top 20 downvotes (-16) score
scores = adhdw['sentiment_scores'][84285]
negative_score = scores['neg']
neutral_score = scores['neu']
positive_score = scores['pos']
compound_score = scores['compound']

# Printing the scores
print("Negative score:", negative_score)
print("Neutral score:", neutral_score)
print("Positive score:", positive_score)
print("Compound score:", compound_score)
print(adhdw['body'][84285])
print("Downvotes:", adhdw['score'][84285])

Negative score: 0.3
Neutral score: 0.7
Positive score: 0.0
Compound score: -0.4588
So am I going to get banned from this sub too for saying I’m going to vote Trump?
Downvotes: -16


In [17]:
# Initialize variables to store cumulative scores
total_negative = 0
total_neutral = 0
total_positive = 0
total_compound = 0
num_documents = len(adhdw)

# Iterate over the sentiment scores and accumulate the scores
for scores_dict in adhdw['sentiment_scores']:
    total_negative += scores_dict['neg']
    total_neutral += scores_dict['neu']
    total_positive += scores_dict['pos']
    total_compound += scores_dict['compound']

# Calculate the average scores
average_negative = total_negative / num_documents
average_neutral = total_neutral / num_documents
average_positive = total_positive / num_documents
average_compound = total_compound / num_documents

# Print the average scores
print("Average Negative Score:", average_negative)
print("Average Neutral Score:", average_neutral)
print("Average Positive Score:", average_positive)
print('Average Compound Score:', average_compound )

Average Negative Score: 0.10907867441699917
Average Neutral Score: 0.6150517127376681
Average Positive Score: 0.2676934934717366
Average Compound Score: 0.32248199478922424


Average Negative Score: 0.10907867441699917

Average Neutral Score: 0.6150517127376681

Average Positive Score: 0.2676934934717366

Average Compound Score: 0.32248199478922424